cmd/utils/flaggroup.go

```diff
@@ -0,0 +1,319 @@
+			DstDynamoDBTableNameFlag,
+			DstDynamoDBRegionFlag,
+			DstDynamoDBIsProvisionedFlag,
+			DstDynamoDBReadCapacityFlag,
+			DstDynamoDBWriteCapacityFlag,
+		},
+	},
+	{
+		Name: "STATE",
+		Flags: []cli.Flag{
+			StateDBCachingFlag,
+			TrieMemoryCacheSizeFlag,
+			TrieBlockIntervalFlag,
+			TriesInMemoryFlag,
+		},
+	},
+	{
+		Name: "CACHE",
+		Flags: []cli.Flag{
+			CacheTypeFlag,
+			CacheScaleFlag,
+			CacheUsageLevelFlag,
+			MemorySizeFlag,
+			CacheWriteThroughFlag,
+			TxPoolStateCacheFlag,
+			TrieCacheLimitFlag,
+		},
+	},
+	{
+		Name: "CONSENSUS",
+		Flags: []cli.Flag{
+			ServiceChainSignerFlag,
+			RewardbaseFlag,
+		},
+	},
+	{
+		Name: "NETWORKING",
+		Flags: []cli.Flag{
+			BootnodesFlag,
+			ListenPortFlag,
+			SubListenPortFlag,
+			MultiChannelUseFlag,
+			MaxConnectionsFlag,
+			MaxPendingPeersFlag,
+			TargetGasLimitFlag,
+			NATFlag,
+			NoDiscoverFlag,
+			RWTimerWaitTimeFlag,
+			RWTimerIntervalFlag,
+			NetrestrictFlag,
+			NodeKeyFileFlag,
+			NodeKeyHexFlag,
+			NetworkIdFlag,
+			BaobabFlag,
+			CypressFlag,
+		},
+	},
+	{
+		Name: "METRICS",
+		Flags: []cli.Flag{
+			MetricsEnabledFlag,
+			PrometheusExporterFlag,
+			PrometheusExporterPortFlag,
+		},
+	},
+	{
+		Name: "VIRTUAL MACHINE",
+		Flags: []cli.Flag{
+			VMEnableDebugFlag,
+			VMLogTargetFlag,
+			VMTraceInternalTxFlag,
+		},
+	},
+	{
+		Name: "API AND CONSOLE",
+		Flags: []cli.Flag{
+			RPCEnabledFlag,
+			RPCListenAddrFlag,
+			RPCPortFlag,
+			RPCCORSDomainFlag,
+			RPCVirtualHostsFlag,
+			RPCApiFlag,
+			IPCDisabledFlag,
+			IPCPathFlag,
+			WSEnabledFlag,
+			WSListenAddrFlag,
+			WSPortFlag,
+			WSApiFlag,
+			WSAllowedOriginsFlag,
+			GRPCEnabledFlag,
+			GRPCListenAddrFlag,
+			GRPCPortFlag,
+			JSpathFlag,
+			ExecFlag,
+			PreloadJSFlag,
+			MaxRequestContentLengthFlag,
+		},
+	},
+	{
+		Name:  "LOGGING AND DEBUGGING",
+		Flags: debug.Flags,
+	},
+	{
+		Name: "SERVICECHAIN",
+		Flags: []cli.Flag{
+			ChildChainIndexingFlag,
+			MainBridgeFlag,
+			MainBridgeListenPortFlag,
+			SubBridgeFlag,
+			SubBridgeListenPortFlag,
+			AnchoringPeriodFlag,
+			SentChainTxsLimit,
+			ParentChainIDFlag,
+			VTRecoveryFlag,
+			VTRecoveryIntervalFlag,
+			ServiceChainAnchoringFlag,
+			ServiceChainNewAccountFlag,
+			KASServiceChainAnchorFlag,
+			KASServiceChainAnchorPeriodFlag,
+			KASServiceChainAnchorUrlFlag,
+			KASServiceChainAnchorOperatorFlag,
+			KASServiceChainAccessKeyFlag,
+			KASServiceChainSecretKeyFlag,
+			KASServiceChainXChainIdFlag,
+		},
+	},
+	{
+		Name: "MISC",
+		Flags: []cli.Flag{
+			GenKeyFlag,
+			WriteAddressFlag,
+			AutoRestartFlag,
+			RestartTimeOutFlag,
+			DaemonPathFlag,
+		},
+	},
+}
+
+// CategorizeFlags classified each flag into pre-defined flagGroups.
+func CategorizeFlags(flags []cli.Flag) []FlagGroup {
+	flagGroupsMap := make(map[string][]cli.Flag)
+	isFlagAdded := make(map[string]bool) // Check duplicated flags
+
+	// Find its group for each flag
+	for _, flag := range flags {
+		if isFlagAdded[flag.GetName()] {
```

Comment:
Isn't it an error case if `isFlagAdded` returns true?

Reply:
I added a debug log. I think it is not an error since it doesn't affect any semantics of Klaytn node and a minor issue.

Reply:
I mean that's something should be checked during the test so that we can avoid that situation, though it does not make any functional or performance issue.

Reply:
I agreed with you. So, I made `TestFlagGroups_Duplication` to avoid the case a flag belongs to more than one group. 

Maybe you also consider the case one flag is added twice with the same values. However, I am considering the case is a normal. Because the input flag slice is a combination of other flag groups having different classification standards(`CommonNodeFlags`, `CommonRPCFlags` and others defined in `cmd/utils/nodecmd/nodeflags.go`. In those standards, a flag can belong to multiple groups.

Reply:
Thanks for the explanation. I understand your point. 😄